editor/filesystem_dock.cpp

```diff
@@ -1852,6 +1907,50 @@ void FileSystemDock::_overwrite_dialog_action(bool p_overwrite) {
 	_move_operation_confirm(to_move_path, to_move_or_copy, p_overwrite ? OVERWRITE_REPLACE : OVERWRITE_RENAME);
 }
 
+void FileSystemDock::_convert_dialog_action() {
+	Vector<String> valid_conversion_targets = _get_valid_conversions_for_file_paths(to_convert);
+
+	Vector<Ref<Resource>> selected_resources;
+	for (const String &S : to_convert) {
+		Ref<Resource> res = ResourceLoader::load(S);
+		ERR_FAIL_COND(!res.is_valid());
+		selected_resources.push_back(res);
+	}
+
+	Vector<Ref<Resource>> converted_resources;
+	for (Ref<Resource> res : selected_resources) {
+		Vector<Ref<EditorResourceConversionPlugin>> conversions = EditorNode::get_singleton()->find_resource_conversion_plugin_for_resource(res);
+		for (Ref<EditorResourceConversionPlugin> E : conversions) {
+			if (E->converts_to() == valid_conversion_targets[selected_conversion_id]) {
+				Ref<Resource> converted_res = E->convert(res);
+				ERR_FAIL_COND(!res.is_valid());
+				converted_resources.push_back(converted_res);
+
+				// Clear history for the objects being replaced.
+				EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+				undo_redo->clear_history(true, undo_redo->get_history_id_for_object(res.ptr()));
+
+				break;
+			}
+		}
+	}
+
+	// Updates all the resources existing as node properties.
+	EditorNode::get_singleton()->replace_resources_in_scenes(selected_resources, converted_resources);
+
+	// Overwrite the old resources.
+	for (int i = 0; i < converted_resources.size(); i++) {
+		Ref<Resource> original_resource = selected_resources.get(i);
+		Ref<Resource> new_resource = converted_resources.get(i);
+
+		// Overwrite the path and UUID.
+		new_resource->set_path(original_resource->get_path(), true);
+		new_resource->set_scene_unique_id(original_resource->get_scene_unique_id());
```

Comment:
Scene unique ID is the ID for resources referenced in a scene. It doesn't make sense in a context of stand-alone external resources.

Both UID and path are automatically handled by `ResourceSaver`, so you can remove these 2 lines.